pyright/_utils.py

```diff
@@ -31,7 +34,7 @@ def install_pyright(args: tuple[object, ...], *, quiet: bool | None) -> Path:
     This accepts a single argument which corresponds to the arguments given to the CLI / langserver
     which are used to determine whether or not certain warnings / logs will be printed.
     """
-    version = os.environ.get('PYRIGHT_PYTHON_FORCE_VERSION', __pyright_version__)
+    version = _get_configured_pyright_version()
     if version == 'latest':
         version = node.latest('pyright')
     else:
```

Comment:
Should I not report this warning when `PYRIGHT_PYTHON_PYLANCE_VERSION` is in use? What do you think?

Reply:
IIRC for context Eric Traut mentioned that he wanted a warning here because he didn't want to get bug reports on the pyright repo for older versions.

So I think keeping a warning here would be a good call but it should be updated to handle the pylance case, e.g. if a version apart from `latest-release` or `latest-prelease` is given *and* there is a new pylance release then the warning should mention that pylance version instead of the current implementation which will mention the latest version published to pypi.

Reply:
This sounds like it could be non-trivial though, so if it's not feasible to get out quickly I don't think it should block this PR but if that is the case I think we should at a minimum disable the version warning for pylance version users

Reply:
Ok, I went with the easier approach of not emitting the existing warning if `PYRIGHT_PYTHON_PYLANCE_VERSION` is provided.